exam_date/stored_date.py

```diff
@@ -0,0 +1,72 @@
+from autologging import logged, traced
+from pathlib import Path
+from datetime import datetime
+
+
+@logged
+@traced
+class AssignmentLatestSubmittedDate():
+    def __init__(self, path: str, file_name: str):
+        self.path = path
+        self.file_name = file_name
+
+    def _create_directory_if_not_exit(self) -> None:
+        """
+        creates a directory if doesn't exit with permission 0o777
+        :return:
+        """
+        try:
+            if not Path(self.path).exists():
+                self.__log.info(f"creating the directory at {self.path}")
+                Path(self.path).mkdir(parents=True, exist_ok=True)
+        except (OSError, Exception) as e:
+            raise e
+
+
+    def get_assign_submitted_date(self) -> str:
+        """
+        creating a directory/file if not available that holds the latest score date. It writes the current UTC date
+        if not provided so that initial run will end up happening.
+        :return: latest_score_timestamp
+        """
+        try:
+            self._create_directory_if_not_exit()
+        except (OSError, Exception) as e:
+            self.__log.error(f"error in creating a directory due to {e}")
+            return None
```

Comment:
The function is defined as returning `str`.  Is it appropriate to return `None` here?  Should it be defined as returning `str` OR `None`?  It's an interesting question, because Python treats `None` as another type, but it can also represent an unset object of a different type.

Reply:
Ya.. this is confusion. The return type is described as `str` and I also return `None`. I felt this needs to raise the exception, the caller function should handle it rather returning `None`. Ya, python treats `None` as `NoneType`
```
>>> type(None)
<class 'NoneType'>
```